### PR TITLE
Add [PypiTypes] badge

### PR DESCRIPTION
**Pull request by yangdanny97**
This PR adds a new badge that signals when a Python package is typed.

It checks Pypi for whether the package has the "Typing :: Typed" classifier, which indicates that the package provides type information via inline type hints or type stub files. This is a useful indicator of the project's compatibility with typecheckers and other developer tooling, and can help increase adoption of type annotations.

Historical context on the "Typing :: Typed" classifier can be found here https://github.com/pypi/warehouse/issues/4348

### CODE REVIEW

**Review by jNullj — CHANGES_REQUESTED** [review]
Looks good, please fix this typo

File: services/pypi/pypi-typing.service.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  static category = 'platform-support'
+
+  static route = this.buildRoute('pypi/typing')
+
+  static openApi = {
+    '/pypi/typing/{packageName}': {
+      get: {
+        summary: 'PyPI - Typing',
+        description:
+          'Whether the package provides type information (inline annotations or stub files)',
```
Comment (chris48s) [inline services/pypi/pypi-typing.service.js:13]: This isn't quite true. What this badge is actually telling us is whether the package metadata contains the classifier `Typing :: Typed`. You can push a package to PyPI that does have annotations but does not use that classifier, and vice versa.

File: services/pypi/pypi-typing.service.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import PypiBase, { pypiGeneralParams } from './pypi-base.js'
+
+export default class PypiTyping extends PypiBase {
+  static category = 'platform-support'
+
+  static route = this.buildRoute('pypi/typing')
```
Comment (chris48s) [inline services/pypi/pypi-typing.service.js:6]: For consistency, the closest other badge to this is the typescript badge, which uses "types" (as opposed to "typing") as the noun part of the route and label. Can we go with `types` here too for both the route and label.

File: services/pypi/pypi-typing.service.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+  static render({ isTyped }) {
+    if (isTyped) {
+      return {
+        message: 'typed',
+        color: 'brightgreen',
+      }
+    } else {
+      return {
+        message: 'no',
```
Comment (chris48s) [inline services/pypi/pypi-typing.service.js:29]: I feel like the words here should be opposites: "yes" and "no", or "typed" and "untyped". "typed" and "no" doesn't seem like the right options.

`types | yes` and `types | no` would be consistent with what we've done on the PyPI Wheel badge
https://github.com/badges/shields/blob/00c73c872d9baa5b3e22c658b8d8b73b8b311f7d/services/pypi/pypi-wheel.service.js#L20-L32 which is conceptually similar

File: services/pypi/pypi-typing.service.js (hunk 1 above)
Comment (yangdanny97) [reply services/pypi/pypi-typing.service.js:13]: I wasn't sure whether to focus more on the intent of the badge or the implementation details here.

"whether or not the package has types" is very complicated, even more so if I think about including standalone stubs packages, typeshed, etc. 

My plan was to have the badge first, and then follow up by making sure that packages have types all have this classifier in the metadata.

There's some context and discussion about the badge here: https://discuss.python.org/t/github-badge-for-type-coverage/74116/

Comment (chris48s) [reply services/pypi/pypi-typing.service.js]: I was really just meaning the badge description should accurately explain what we're actually doing/showing.

It is outside the scope of what we can do on a badge to actually perform static analysis on a package in the context of a request/response cycle, but if there were a service that did this and exposed type coverage/whether the types for a package are inline or stubs/etc for a given package, we could certainly consider adding badges that consume that data in future. Similar to how we have PyPI downloads badges using the data from https://pypistats.org/

**Review by chris48s — APPROVED** [review]

**Pull request merged**